### PR TITLE
SCRUM-2600: Upgrade to Node 18

### PR DESCRIPTION
**Pull request by abecerra**
- Adding babel/plugin-proposal-private-property-in-object silences a long warning in the UI tests, maybe it can be removed later in a ticket to upgrade/remove dependencies

### CODE REVIEW

**Review by oblodgett — DISMISSED** [review]
This does not compile locally:

```
Compiled with problems:
×
ERROR in ./src/components/GenericDataTable/GenericDataTable.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: [BABEL] /Users/olinblodgett/git/agr_curation/src/main/cliapp/src/components/GenericDataTable/GenericDataTable.js: --- PLACEHOLDER PACKAGE ---
This @babel/plugin-proposal-private-property-in-object version is not meant to
be imported. Something is importing
@babel/plugin-proposal-private-property-in-object without declaring it in its
dependencies (or devDependencies) in the package.json file.
Add "@babel/plugin-proposal-private-property-in-object" to your devDependencies
to work around this error. This will make this message go away.
```

**Comment by oblodgett** [comment]
This also disrupts all local development for the UI and this story would be better pulled in in the off release.

File: src/main/cliapp/package.json
Comment (oblodgett) [inline src/main/cliapp/package.json]: Plus the indention is wrong with this.

Comment (mluypaert) [reply src/main/cliapp/package.json]: To add in to this, this version reference looks like a development version, which should not be used for importing in our project (use stable versions only). The current version on npm is `7.21.11` so I would suggest using that.

File: src/main/cliapp/package.json
Comment (mluypaert) [inline src/main/cliapp/package.json]: `babel/plugin-proposal-private-property-in-object` must be imported at a production-grade version (I'd propose `^7.21.11`), instead of the development version used here.

**Review by mluypaert — CHANGES_REQUESTED** [review]

File: src/main/cliapp/package.json
Comment (mluypaert) [inline src/main/cliapp/package.json]: Same here, production version please.

**Review by mluypaert — CHANGES_REQUESTED** [review]

File: src/main/cliapp/package.json
Comment (abecerra) [reply src/main/cliapp/package.json]: set in last commit

File: src/main/cliapp/package.json
Comment (abecerra) [reply src/main/cliapp/package.json]: set in last commit

File: src/main/cliapp/package.json
Comment (abecerra) [reply src/main/cliapp/package.json]: set in last commit

**Comment by mluypaert** [comment]
> This does not compile locally:
> 
> ```
> Compiled with problems:
> ×
> ERROR in ./src/components/GenericDataTable/GenericDataTable.js
> Module build failed (from ./node_modules/babel-loader/lib/index.js):
> Error: [BABEL] /Users/olinblodgett/git/agr_curation/src/main/cliapp/src/components/GenericDataTable/GenericDataTable.js: --- PLACEHOLDER PACKAGE ---
> This @babel/plugin-proposal-private-property-in-object version is not meant to
> be imported. Something is importing
> @babel/plugin-proposal-private-property-in-object without declaring it in its
> dependencies (or devDependencies) in the package.json file.
> Add "@babel/plugin-proposal-private-property-in-object" to your devDependencies
> to work around this error. This will make this message go away.
> ```

@oblodgett What version of Node are you using? Seems to compile fine on my local system using Node `v18.14.1`

**Comment by oblodgett** [comment]
Yes it probably works on the correct node version but this update now requires all developers to stop and update node. Again this needs to wait till the off release sprint.

**Comment by abecerra** [comment]
> This also disrupts all local development for the UI and this story would be better pulled in in the off release.

To merge it needs in local environments:

- delete package-lock.json node_modules
- nvm use v18.16.0 
- npm install

So it could be merged tomorrow morning (UK)

**Comment by oblodgett** [comment]
No... the team did not agree to pull in this story.

**Comment by mluypaert** [comment]
I disagree that this can only get merge during off-release (updating your node version for this project really is a trivial task if you're using `nvm`), but I agree that this tickets need approval by the team before merge.

**Comment by oblodgett** [comment]
@markquintontulloch and @adamgibs are the primary UI developers, they are the ones that need to drive this ticket. If they are fine with it and can get the right version of node installed sure bring it in, but we did not discuss this as a team.

**Comment by adamgibs** [comment]
@oblodgett @abecerra @mluypaert @markquintontulloch This seems to work for me locally. Component test seem to run as well.

**Comment by markquintontulloch** [comment]
I'm fine with pulling it in, but agree that the timing of breaking changes to the development environment should really be discussed with the team in advance.

**Review by adamgibs — APPROVED** [review]

**Comment by abecerra** [comment]
> I'm fine with pulling it in, but agree that the timing of breaking changes to the development environment should really be discussed with the team in advance.

Yes, but the PR helps testing the github actions changes. Once approved the merging can wait discussion of course.

**Review by mluypaert — APPROVED** [review]

**Pull request merged**
